lib/nexpose/vulnerability.rb

```diff
@@ -108,5 +96,47 @@ def method_missing(method, *args)
 
       nil
     end
+
+    private
+
+    def cleanup_nested(source)
+      result = source.to_s
+      result.gsub!(/<references>/, '')
+      result.gsub!(/<\/references>/, '')
+      result.gsub!(/<reference source=\"(.*?)\">(.*?)<\/reference>/i) {"#{$1.strip}: #{$2.strip}\n"}
+      result.gsub!(/<tags>/, '')
+      result.gsub!(/<\/tags>/, '')
+      result.gsub!(/<tag>(.*?)<\/tag>/) {"#{$1}\n"}
+      result.gsub!(/        /, '')
+      result
+    end
+
+    def cleanup_html(source)
+      result = source.to_s
+      result.gsub!(/<ContainerBlockElement>/, '')
+      result.gsub!(/<\/ContainerBlockElement>/, '')
+      result.gsub!(/<Paragraph>(.*?)<\/Paragraph>/, '\1')
+      result.gsub!(/<Paragraph preformat=\"true\">(.*?)<\/Paragraph>/m){|m| "\nbc. #{ $1 }\n\n"}
+      result.gsub!(/<Paragraph>/, '')
```

Comment:
@rachkor is the goal of this line to capture `<Paragraph>` with newline in between? If yes, then you can update L#118 to capture with newline too. Instead of using `.` which captures all except newline, you can use `[\s\S]` which captures all characters. You can reduce `<UnorderedList>` and `<ListItem>` to one line with this.

Reply:
@aapomm I pushed a new commit to clean up the Regex, thank you! I still had to keep the extra lines to capture trailing `<Paragraph>` and `</Paragraph>` tags. They're sort of randomly nested in the NeXpose output so even with new lines captured, `Paragraph` tags remained. But, the others (list-related tags) are optimized now!